scipy/linalg/__init__.py

```diff
@@ -38,6 +38,7 @@
    tril - Construct a lower-triangular matrix from a given matrix
    triu - Construct an upper-triangular matrix from a given matrix
    orthogonal_procrustes - Solve an orthogonal Procrustes problem
+   balance - Balance matrix entries with a diagonal similarity transformation
```

Comment:
This adds a new public name, right? Something more specific would be nice, simple "balance" sounds too generic.